### PR TITLE
Refactor decomp edit files + SceneExport

### PR DESCRIPTION
**Pull request by kurethedead**
Refactored scene_table.py to move unnecessary stored state from classes into parameters for functions on the classes. Also tried to avoid passing SceneExport down multiple functions, and instead only passing down necessary arguments. Converted old scene deletion code.

### CODE REVIEW

**Comment by kurethedead** [comment]
I've also refactored SceneExport so that it only has static methods, with all its previous params being aggregated into ExportInfo. This removes a lot of redundant argument passing and hopefully makes the arguments more understandable.

File: fast64_internal/oot/exporter/__init__.py [hunk 1]
```diff
@@ -1 +1,123 @@
...
+    RemoveInfo,
+    OOTObjectCategorizer,
+    ootDuplicateHierarchy,
+    ootCleanupScene,
+    getSceneDirFromLevelName,
+    ootGetPath,
+)
+
+
+@dataclass
```
Comment (Yanis002) [inline fast64_internal/oot/exporter/__init__.py]: this is probably not necessary anymore (the corresponding imports too)

File: fast64_internal/oot/exporter/decomp_edit/scene_table.py [hunk 2]
```diff
@@ -1,312 +1,289 @@
...
+
+        lines = data[header_end_index + 1 :].split("\n")
+        lines = list(filter(None, lines))  # removes empty lines
+        lines = [line.strip() for line in lines]
+
+        sections: list[SceneTableSection] = []
+        current_section: Optional[SceneTableSection] = None
+
+        while len(lines) > 0:
+            line = lines.pop(0)
```
Comment (Yanis002) [inline fast64_internal/oot/exporter/decomp_edit/scene_table.py]: why not ``for line in lines``? looks like ``lines`` isn't used after this

File: fast64_internal/oot/oot_utility.py [hunk 3]
```diff
@@ -10,6 +10,7 @@
 from bpy.types import Object
 from typing import Callable, Optional
 from .oot_constants import ootSceneIDToName
+from dataclasses import dataclass, field
```
Comment (Yanis002) [inline fast64_internal/oot/oot_utility.py]: looks like ``field`` isn't used?

**Review by Yanis002 — COMMENTED** [review]
nice

**Pull request merged**